x-pack/plugins/cloud_security_posture/server/lib/telemetry/collectors/types.ts

```diff
@@ -8,7 +8,7 @@
 import { CspStatusCode } from '../../../../common/types';
 
 export interface CspmUsage {
-  indices: CspmIndicesStats;
+  indices: CspmIndicesStats | never[];
```

Comment:
Why are we using `never` here?

Reply:
When I added, the empty array in case Promise fails, then I had to add the type for `never[]`  which means there are never values in an array. That's the incorrect error handling so if a promising value is rejected then  I change the return value to be  `undefined`. Now  the type will change to   `CspmIndicesStats |  undefined`